test/unit/modunit_pico_6lowpan.c

```diff
@@ -0,0 +1,1616 @@
+            size = 64; /* Default */
+        }
+    }
+
+    if (next) {
+        next = cpy_arg(&abort, next);
+        if (strlen(abort) > 0) {
+            printf("Got arg: '%s'\n", abort);
+            timeout = atoi(abort);
+            if (timeout < 0) {
+                fprintf(stderr, "ping needs the following format: ping:dst_addr:[size:[abort after N sec:[wait N sec before start]]]\n");
+                exit(255);
+            }
+            printf("Aborting ping after %d seconds\n", timeout);
+        }
+    }
+
+    if (next) {
+        next = cpy_arg(&delay, next);
+        if (strlen(delay) > 0) {
+            initial_delay = atoi(delay);
+            if (initial_delay > 0) {
+                printf("Initial delay: %d seconds\n", initial_delay);
+                initial_delay = PICO_TIME_MS() + initial_delay * 1000;
+                while (PICO_TIME_MS() < initial_delay) {
+                    pico_stack_tick();
+                    usleep(10000);
+                }
+            }
+        }
+    }
+    printf("Starting ping.\n");
+
+    id = pico_icmp6_ping(dest, NUM_PING, 1000, 10000, size, cb_ping6, NULL);
+    if (timeout > 0) {
+        printf("Adding abort timer after %d seconds for id %d\n", timeout, id);
+        if (!pico_timer_add(timeout * 1000, ping_abort_timer, &id)) {
+            printf("Failed to set ping abort timeout, aborting ping\n");
+            ping_abort_timer((pico_time)0, &id);
+            exit(1);
+        }
+    }
+
+    /* free copied args */
+    if (dest)
+        free(dest);
+
+    if (abort)
+        free(abort);
+}
+
+START_TEST(tc_tx_rx)
+{
+    int test = 0;
+    struct pico_device *dev = NULL;
+    uint8_t n_id, n_area0, n_area1;
+    struct pico_ip6 myaddr, pan, netmask;
+    const char pan_addr[] = "2aaa:abcd::0";
+    const char pan_netmask[] = "ffff:ffff:ffff:ffff::0";
+
+    char *id = "3";
+    char *area0 = "1";
+    char *area1 = "0";
+    char *dump = "build/test/unit_6lowpan.pcap";
+    char *arg = strdup("2aaa:abcd:0000:0000:0200:00aa:ab00:0001,1450,0,1,");
+
+    STARTING();
+
+    n_id = (uint8_t) atoi(id);
+    n_area0 = (uint8_t) atoi(area0);
+    n_area1 = (uint8_t) atoi(area1);
+
+    /* Initialize picoTCP */
+    pico_stack_init();
+
+    pico_string_to_ipv6(pan_addr, myaddr.addr);
+    pico_string_to_ipv6(pan_addr, pan.addr);
+    pico_string_to_ipv6(pan_netmask, netmask.addr);
+    myaddr.addr[8]  = 0x02;
+    myaddr.addr[11] = 0xaa;
+    myaddr.addr[12] = 0xab;
+    myaddr.addr[15] = n_id;
+
+    printf("%d:%d:%d\n", n_id, n_area0, n_area1);
+    dev = pico_radiotest_create(n_id, n_area0, n_area1, 1, dump);
+    if (!dev) {
+        exit(1);
+    }
+
+    printf("Radiotest created.\n");
+
+    /* Add a routable link */
+    pico_ipv6_link_add(dev, myaddr, netmask);
+
+    /* Start ping-application */
+    app_ping(arg);
+
+    printf("%s: launching PicoTCP loop\n", __FUNCTION__);
+    while(!tx || !rx) {
+        pico_stack_tick();
+        usleep(2000);
+    }
```

Comment:
Not sure this still counts as a *unit* test. (Note : this doesn't mean it's not a valuable test!)
Maybe @ISY-Thomas can have a look and give his opinion? Everything from line 1344 down to about here is all related to this one test case...